scripts/validate_string_concatenation.py

```diff
@@ -102,28 +106,157 @@ def strings_to_concatenate(source_path: str) -> Generator[Tuple[str, int], None,
         Source file path.
     line_number : int
         Line number of unconcatenated string.
+    MSG : str
+        Explenation of the error.
+
+    Notes
+    -----
+    GH #30454
     """
+    MSG: str = (
+        "String unnecessarily split in two by black. Please merge them manually."
+    )
+
     with open(source_path, "r") as file_name:
         tokens: List = list(tokenize.generate_tokens(file_name.readline))
 
     for current_token, next_token in zip(tokens, tokens[1:]):
         if current_token[0] == next_token[0] == token.STRING:
-            yield source_path, current_token[2][0]
+            yield source_path, current_token[2][0], MSG
+
+
+def strings_with_wrong_placed_space(
+    source_path: str,
+) -> Generator[Tuple[str, int, str], None, None]:
+    """
+    Test case for leading spaces in concated strings.
+
+    For example:
+
+    >>> foo = (
+    ...    "bar "
+    ...    "baz"
+    ... )
+
+    Instead of:
+
+    >>> foo = (
+    ...        "bar"
+    ...       " baz"
```

Comment:
This indentation is different than the previous, and makes it more difficult to compare. Also, if instead of `foo`, `bar`... you explain what's going on, the same example can be clearer. e.g.:
```python
rule = (
    "We want the space at the end of lines, "
    "not at the beginning"
)
```